### ISSUE TITLE
IDE Fatal Error: NullPointerException

### ISSUE THREAD

**Issue by smcoll** [Ashald/EnvFile#14]
PyCharm 2016.1.4 reported a fatal error.  The notice came when i opened a second IDE window, but i don't now if that was when it occurred, or if that was when i was notified: `Exception in plugin Env File. Today at 9:19 AM. Occurred 2 times since the last clear. Unread.`

This is all the context i have, unfortunately.

```
null
java.lang.NullPointerException
    at net.ashald.envfile.PyEnvFileRunConfigurationExtension.patchCommandLine(PyEnvFileRunConfigurationExtension.java:104)
    at net.ashald.envfile.PyEnvFileRunConfigurationExtension.patchCommandLine(PyEnvFileRunConfigurationExtension.java:24)
    at com.intellij.execution.configuration.RunConfigurationExtensionsManager.patchCommandLine(RunConfigurationExtensionsManager.java:160)
    at com.jetbrains.python.run.PythonCommandLineState.startProcess(PythonCommandLineState.java:190)
    at com.jetbrains.django.run.DjangoServerRunCommandLineState.startProcess(DjangoServerRunCommandLineState.java:206)
    at com.jetbrains.python.run.PythonCommandLineState.execute(PythonCommandLineState.java:131)
    at com.jetbrains.python.run.PythonRunner.doExecute(PythonRunner.java:51)
    at com.intellij.execution.runners.GenericProgramRunner$1.execute(GenericProgramRunner.java:43)
    at com.intellij.execution.impl.ExecutionManagerImpl$5.run(ExecutionManagerImpl.java:389)
    at com.intellij.execution.impl.ExecutionManagerImpl.compileAndRun(ExecutionManagerImpl.java:291)
    at com.intellij.execution.impl.ExecutionManagerImpl.startRunProfile(ExecutionManagerImpl.java:432)
    at com.intellij.execution.runners.GenericProgramRunner.execute(GenericProgramRunner.java:40)
    at com.intellij.execution.runners.BaseProgramRunner.execute(BaseProgramRunner.java:61)
    at com.intellij.execution.runners.GenericProgramRunner.execute(GenericProgramRunner.java:31)
    at com.intellij.execution.runners.BaseProgramRunner.execute(BaseProgramRunner.java:50)
    at com.intellij.execution.runners.GenericProgramRunner.execute(GenericProgramRunner.java:31)
    at com.intellij.execution.ProgramRunnerUtil.executeConfiguration(ProgramRunnerUtil.java:121)
    at com.intellij.execution.impl.ExecutionManagerImpl.a(ExecutionManagerImpl.java:116)
    at com.intellij.execution.impl.ExecutionManagerImpl.access$400(ExecutionManagerImpl.java:67)
    at com.intellij.execution.impl.ExecutionManagerImpl$7.run(ExecutionManagerImpl.java:516)
    at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
    at com.intellij.util.Alarm$Request$1.run(Alarm.java:378)
    at com.intellij.util.Alarm$Request.run(Alarm.java:398)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:227)
    at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
    at java.awt.EventQueue.access$500(EventQueue.java:97)
    at java.awt.EventQueue$3.run(EventQueue.java:709)
    at java.awt.EventQueue$3.run(EventQueue.java:703)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
    at com.intellij.ide.IdeEventQueue.g(IdeEventQueue.java:857)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:658)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:386)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by ashald** [comment]
Thanks for the report - I'll look into it and will try to release fix soon.

**Comment by smcoll** [comment]
@Ashald i ran into this again today.  Do you need anything to help debug?

**Comment by ashald** [comment]
@smcoll I tried to reproduce an error but without any success. Can you share more details what actions lead to it?

**Comment by ashald** [comment]
BTW, are you running CE or PE version?

**Comment by smcoll** [comment]
Good news: figured out how to replicate the issue.  The selector for the type of file (env/yaml) next to the file selector was somehow empty.  Once i selected `.env` the issue sorted itself out.  i don't know where the empty option came from; maybe that was an artifact from an earlier version of the plugin?

i'm on 2016.2.1 PE now.

**Comment by ashald** [comment]
Unfortunately I wasn't able to reproduce the issue. Will appreciate if people can comment whether they still experience the issue and whether there is a sure way to reproduce it.

**Comment by underyx** [comment]
@Ashald happened to me today. I think the latest PyCharm update (which I installed yesterday night) somehow cleared my `.env` selection for the file type setting?

It might be enough to just add some graceful handling to that field being empty.

**Comment by ashald** [comment]
This should be fixed in `v2.0` so closing the issue.
